kubespawner/spawner.py

```diff
@@ -1945,7 +1946,16 @@ def _get_pod_url(self, pod):
                 ]
             )
 
-        return "{}://{}:{}".format(
+        fmtstr = "{}://{}:{}"
+        # Parsing as an IP Address may fail if hostname is a DNS name.
+        try:
+            addr = ipaddress.ip_address(hostname)
```

Comment:
I think rather than try/excepting all possible hostnames, the place for this is in the `podIP` branch above on L1936. Then it's a simpler check `if ":" in hostname: hostname = f"[{hostname}]"`

Reply:
Okay, but I think we should still use the `ip_address` parsing thing, just because it's best practice. For example, did you know `::192.168.1.1` is a valid IPv6 address? Admittedly it still has a `:` in it, but... I think we should just throw this problem to the Python standard library.

```python
>>> ipaddress.ip_address("::192.168.1.1")
IPv6Address('::c0a8:101')
```